src/textual/widgets/_markdown.py

```diff
@@ -118,6 +119,95 @@ async def action_link(self, href: str) -> None:
         """Called on link click."""
         self.post_message(Markdown.LinkClicked(self._markdown, href))
 
+    def notify_style_update(self) -> None:
+        """If CSS was reloaded, try to rebuild this block from its token."""
+        super().notify_style_update()
+        self.rebuild()
+
+    def rebuild(self) -> None:
+        """Rebuild the content of the block if we have a source token."""
+        if self._token is not None:
+            self.build_from_token(self._token)
+
+    def build_from_token(self, token: Token) -> None:
+        """Build the block content from its source token.
+
+        This method allows the block to be rebuilt on demand, which is useful
+        when the styles assigned to the
+        [Markdown.COMPONENT_CLASSES][textual.widgets.Markdown.COMPONENT_CLASSES]
+        change.
+
+        See https://github.com/Textualize/textual/issues/3464 for more information.
+
+        Args:
+            token: The token from which this block is built.
+        """
+
+        self._token = token
+        style_stack: list[Style] = [Style()]
```

Comment:
From `style_stack: ...` to the end of this method, I copied and pasted the code.
The only change happens when I'm getting the component classes.
Instead of getting them from `self`, I get them from `self._markdown`.